src/translate/generator.py

```diff
@@ -1,367 +1,378 @@
+                if platform.system() == 'Windows':
+                    _openvr_lib_name = "libopenvr_api_32"
+                elif platform.system() == 'Linux':
+                    _openvr_lib_name = "libopenvr_api_32.so"
+                elif platform.system() == 'Darwin':
+                    _openvr_lib_name = "libopenvr_api_32.dylib"    
+                else:
+                    raise ValueError("Libraries not available for this platform: " + platform.system())
+            else:
+                if platform.system() == 'Windows':
+                    _openvr_lib_name = "libopenvr_api_64"
+                elif platform.system() == 'Linux':
+                    _openvr_lib_name = "libopenvr_api_64.so"
+                elif platform.system() == 'Darwin':
+                    _openvr_lib_name = "libopenvr_api_32.dylib"  # Universal 32-bit and 64-bit library.
+                else:
+                    raise ValueError("Libraries not available for this platform: " + platform.system())
+            
+            # Load library
+            if platform.system() == 'Windows':
+                # Add current directory to PATH, so we can load the DLL from right here.
+                os.environ['PATH'] += os.pathsep + os.path.dirname(__file__)
+            else:
+                _openvr_lib_name = os.path.join(os.path.dirname(__file__), _openvr_lib_name)
+            
+            _openvr = cdll.LoadLibrary(_openvr_lib_name)
+            
+            # Function pointer table calling convention
+            if platform.system() == 'Windows':
+                OPENVR_FNTABLE_CALLTYPE = WINFUNCTYPE  # __stdcall in openvr_capi.h
+            else:
+                OPENVR_FNTABLE_CALLTYPE = CFUNCTYPE  # __cdecl
+            
+            
+            def byref(arg):
+                if arg is None:
+                    return None
+                else:
+                    return ctypes.byref(arg)
+        ''')
+        print(preamble, file=file_out)
+
+    @staticmethod
+    def generate_errors(declarations, file_out):
+        print(inspect.cleandoc('''
+            ############################################
+            # Python exceptions for openvr error codes #
+            ############################################
+
+
+            class OpenVRError(Exception):
+                """
+                OpenVRError is a custom exception type for when OpenVR functions return a failure code.
+                Such a specific exception type allows more precise exception handling that does just raising Exception().
+                """
+                pass
+            
+            
+            class ErrorCode(OpenVRError):
+                error_index = dict()
+                is_error = True  # FooError_None classes should override this
+            
+                @classmethod
+                def check_error_value(cls, error_value, message=''):
+                    error_class = cls.error_index[int(error_value)]
```

Comment:
I'm not sure. Maybe if the `error_value` were still a `ctypes.c_int16` or whatever, it might or might not function correctly as an array index. If it works correctly without the int constructor we can remove it. Otherwise in general I'm inclined to leave mysterious things like that in place, on the assumption that at one time I understood that it was needed.

Reply:
I was probably interested in this "limit" case. On my install, I did not need to cast the value, so I thought maybe on some others (or in different Python versions), one had to. Anyway, it was just a curiosity.